### PR TITLE
create/update and delete tasks/tags

### PR DESCRIPTION
**Pull request by ManonTfx**

### CODE REVIEW

**Review by ThomasBarrial — APPROVED** [review]

File: src/API/mutation/createTask.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { gql } from '@apollo/client';
+
+export const CREATE_TASK = gql`
+  mutation Mutation(
```
Comment (louisledoux) [inline src/API/mutation/createTask.ts]:
```suggestion
export const CREATE_TASK_WITH_TAGS = gql`
  mutation CreateTaskWithTags(
```

File: src/views/tasks/CreateUpdateTask.tsx [hunk 2]
```diff
@@ -1,36 +1,229 @@
...
-import { useForm } from 'react-hook-form';
-import close from '../../assets/icons/close.svg';
+import { useMutation, useQuery } from '@apollo/client';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { toast } from 'react-toastify';
+import OneTag from '@components/tag/OnTag';
+import close from '@assets/icons/close.svg';
+import { format } from 'date-fns';
+import { CREATE_TASK, UPDATE_TASK } from '../../API/mutation/createTask';
```
Comment (louisledoux) [inline src/views/tasks/CreateUpdateTask.tsx]:
```suggestion
import { CREATE_TASK_WITH_TAGS, UPDATE_TASK } from '../../API/mutation/createTask';
```

File: src/views/tasks/CreateUpdateTask.tsx [hunk 3]
```diff
@@ -1,36 +1,229 @@
...
+  const { handleSubmit, register, setValue } = useForm();
+  const [dataProjects, setDataProjects] = useState<IProjectList[]>([]);
+  const [dataTagsList, setDataTagsList] = useState<ITagList[]>([]);
+  const [isModalTag, setIsModalTag] = useState(false);
+  const [dataTag, setDataTag] = useState<ITagList[]>([]);
+
+  // CREATE A TASK
+  const [create, { loading, error }] = useMutation<{
+    createTask: ITaskList;
+  }>(CREATE_TASK, {
```
Comment (louisledoux) [inline src/views/tasks/CreateUpdateTask.tsx]:
```suggestion
  }>(CREATE_TASK_WITH_TAGS, {
```

File: src/API/mutation/createTask.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    $estimeeSpentTime: Float!
+  ) {
+    updateTaskById(
+      id: $updateTaskByIdId
+      subject: $subject
+      projectId: $projectId
+      endDate: $endDate
+      advancement: $advancement
+      estimeeSpentTime: $estimeeSpentTime
+    ) {
```
Comment (louisledoux) [inline src/API/mutation/createTask.ts]:
```suggestion
  mutation UpdateTaskById(
    $taskId: String!
    $subject: String!
    $projectId: String!
    $endDate: String!
    $advancement: String!
    $estimeeSpentTime: Float!
  ) {
    updateTaskById(
      id: $taskId
      subject: $subject
      projectId: $projectId
      endDate: $endDate
      advancement: $advancement
      estimeeSpentTime: $estimeeSpentTime
    ) {
```

File: src/API/mutation/deleteTasks.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { gql } from '@apollo/client';
+
+export const DELETE_TASK = gql`
+  mutation Mutation($deleteTaskByIdId: String!) {
+    deleteTaskById(id: $deleteTaskByIdId) {
+      id
+    }
+  }
```
Comment (louisledoux) [inline src/API/mutation/deleteTasks.ts]:
```suggestion
export const DELETE_TASK = gql`
  mutation DeleteTaskById($taskd: String!) {
    deleteTaskById(id: $taskId) {
      id
    }
  }
```

File: src/API/queries/projectQueries.ts [hunk 6]
```diff
@@ -12,7 +12,7 @@ export const GET_ALL_PROJECTS = gql`
     }
   }
 `;
-export const GetOneProject = gql`
+export const GET_ONE_PROJECT = gql`
   query getProjectByIdId($getProjectByIdId: String!) {
     getProjectByID(id: $getProjectByIdId) {
```
Comment (louisledoux) [inline src/API/queries/projectQueries.ts]:
```suggestion
  query getProjectByIdId($projectId: String!) {
    getProjectByID(id: $projectId) {
```

File: src/API/queries/tagQueries.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import { gql } from '@apollo/client';
+
+export const GET_ALL_TAGS = gql`
+  query Query {
```
Comment (louisledoux) [inline src/API/queries/tagQueries.ts]:
```suggestion
  query GetAllTags {
```

File: src/API/queries/tagQueries.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    getAllTags {
+      id
+      label
+      color
+    }
+  }
+`;
+export const GET_ONE_TAG = gql`
+  query GetTagByID($getTagByIdId: String!) {
+    getTagByID(id: $getTagByIdId) {
```
Comment (louisledoux) [inline src/API/queries/tagQueries.ts]:
```suggestion
  query GetTagByID($tagId: String!) {
    getTagByID(id: $tagId) {
```

File: src/API/queries/taskQueries.ts [hunk 9]
```diff
@@ -9,11 +9,16 @@ export const GetAllTasks = gql`
...
+        color
+      }
     }
   }
 `;
 
-export const GetOneTask = gql`
+export const GET_ONE_TASK = gql`
   query getTaskByID($getTaskByIdId: String!) {
     getTaskByID(id: $getTaskByIdId) {
```
Comment (louisledoux) [inline src/API/queries/taskQueries.ts]:
```suggestion
  query getTaskByID($taskId: String!) {
    getTaskByID(id: $taskId) {
```

File: src/index.d.ts [hunk 10]
```diff
@@ -60,6 +71,18 @@ interface getTaskByID {
...
   endDate: string;
   estimeeSpentTime: string;
+  tags: ITagList;
+}
+interface ITagList {
+  __typename?: string;
+  id: string;
+  label: string;
+  color: string;
+}
```
Comment (louisledoux) [inline src/index.d.ts]: Soon you'll disappear... and I love it!

f6d56508e8f10cce95a752bd1dc38613

File: src/router.tsx [hunk 11]
```diff
@@ -7,8 +7,8 @@ import Settings from '@views/Settings';
 import TaskList from '@views/tasks/TaskList';
 import UsersList from '@views/UsersList';
 import Page404 from '@views/Page404';
-import ProjectDetails from '@views/projects/ProjectDetails';
 import TaskDetails from '@views/tasks/TaskDetails';
+import ProjectDetails from './views/projects/ProjectDetails';
```
Comment (louisledoux) [inline src/router.tsx:11]:
```suggestion
import ProjectDetails from '@views/projects/ProjectDetails';
```

**Review by louisledoux — CHANGES_REQUESTED** [review]

**Review by louisledoux — APPROVED** [review]

**Pull request merged**